### PR TITLE
Tweak/better differentiation service options 1909

### PR DESCRIPTION
**Pull request by Ferdev**
Implements the new design defined in #1909.

This is the list of tasks needed to implement the new design:

- [x] Adapt current unit test setup to support components testing.
- [x] Create unit test for the `ShippingRate` component.
- [x] Add test for the existing and the new uses cases.
- [x] Add the benefits list
- [x] Implement action to toggle the signature options section when a new rate is selected.
- [x] Transform the current signature combo box to a list of checkboxes.
- [x] Make the checkboxes behave as radio buttons.
- [x] Update the rate's amount when a signature option is selected.
- [x] Update the shipping label totals when a signature option is selected.

### CODE REVIEW

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/shipping-rate.js [hunk 1]
```diff
@@ -19,45 +21,79 @@ class ShippingRate extends Component {
...
+	renderServices( carrier_id, signatureOptions, includedServices ) {
+
+		const servicesToRender = [];
 
-		this.setState( { signatureOption: val } );
-		updateValue( service_id, val );
+		if ( includedServices.tracking ) {
+			switch ( carrier_id ) {
+				case 'usps':
+					// Ideally this would come from connect-server, but we have no info from EasyPost API
```
Comment (superdav42) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/shipping-rate.js]: I think we could get rid of this comment now. Tracking will be coming from the connect-server now and we'll only show this on services that support it. I guess we can keep the switch statement as it'll say what kind of tracking it uses. It seems a little redundant to say USPS tracking as USPS offers no other kind. I think we could just say "Includes tracking"  and not use a switch statement.

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/shipping-rate.js [hunk 2]
```diff
@@ -81,41 +118,31 @@ class ShippingRate extends Component {
...
-				// Refer to: https://www.easypost.com/docs/api/node#trackers, specifically
-				// `A Tracker is created automatically whenever you buy a Shipment through EasyPost`
-				details = translate( 'Includes USPS tracking' );
-				break;
-		}
+		const ratePlusSignatureCost = selectedSignature ? rate + selectedSignature.value : rate;
+
 		return(
-			<div className="rates-step__shipping-rate-container">
+			<div className="rates-step__shipping-rate-container" >
```
Comment (superdav42) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/shipping-rate.js]:
```suggestion
			<div className="rates-step__shipping-rate-container">
```

**Comment by superdav42** [comment]
Also the tooltip after "Available for free pickup" is on the design but does not seem to be implemented. Was that intentional?

**Comment by Ferdev** [comment]
This can be reviewed again, thanks @superdav42 !

**Comment by Ferdev** [comment]
To test it with the latest API, grab this WC Connect Server PR https://github.com/Automattic/woocommerce-connect-server/pull/1353 too

File: classes/class-wc-connect-api-client.php [hunk 3]
```diff
@@ -466,7 +466,7 @@ protected function request_headers() {
 			$lang = $locale_elements[ 0 ];
 			$headers[ 'Accept-Language' ] = $locale . ',' . $lang;
 			$headers[ 'Content-Type' ] = 'application/json; charset=utf-8';
-			$headers[ 'Accept' ] = 'application/vnd.woocommerce-connect.v1';
+			$headers[ 'Accept' ] = 'application/vnd.woocommerce-connect.v2';
```
Comment (superdav42) [inline classes/class-wc-connect-api-client.php]: The v2 is kinda like a magic number. I think we should put it in a constant. This is how I had it in my PoC. You should be able to rebase from https://github.com/Automattic/woocommerce-services/pull/1913/files then it'll be more obvious how to update to a newer API later.

File: assets/stylesheets/style.scss [hunk 4]
```diff
@@ -238,6 +238,9 @@
 @import 'components/popover/style';
 @import 'components/tooltip/style';
 
+// All WordPress Component styles
+@import '~@wordpress/components/build-style/style.css';
```
Comment (superdav42) [inline assets/stylesheets/style.scss]: Not sure why this change is showing here. It's already in develop, I added as part of my tooltip work, did you add it too? Probably you should merge develop and make sure this line is not duplicated.

Comment (Ferdev) [reply assets/stylesheets/style.scss]: Yeah, I rebased my branch off yours so the merge will be easier. It shouldn't cause any problems.

**Comment by Ferdev** [comment]
Hey @superdav42 , please, give it another try. The tooltip not being displayed is actually caused by wc-admin. If you disable the plugin the notifications appear again. I've created this issue so we don't forget to fix it https://github.com/Automattic/woocommerce-services/issues/1951

**Pull request merged**